### PR TITLE
Update azure-openai to check OIDC TokenCredential

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR is similar to #626. 

If #626 is approved then, as far as this PR is concerned, once the azure-openai and openai-common changes are reviewed, I'd trim it by avoiding adding yet another demo identical to the one added in #626 and instead, try to merge both demos into a single one, showing multiple quarkus-langchain4j models in action and complementing it with annotation based OIDC tenant selection (either Google or Microsoft Entra ID)

### CODE REVIEW

**Comment by geoand** [comment]
Thanks!

I'll have a look next week,  but I want @csotiriou to also have a look at this

**Comment by sberyozkin** [comment]
Thanks @geoand, @csotiriou. Agreed, #646 is great. My hope with this PR is that it just works for Quarkus OIDC or SmallRye JWT users, they only configure Azure OIDC provider, and Azure OpenAI provider and is done, though asking users register dynamic filters to have OIDC tokens propagated can be quite reasonable too.

But in any case, for this PR, I'm yet to confirm I can login to Azure and use my authorization code flow token to access Azure OpenAI, I did a few tries yesterday, but to no avail. 

@csotiriou If you know what scope I should set [here](https://github.com/quarkiverse/quarkus-langchain4j/pull/634/files#diff-2f75469ca5e3e4495cd9e80c22d1b620177f7af6fad97896466c301955f8ecadR6) to make it work then let me know please :-)

**Comment by csotiriou** [comment]
> But in any case, for this PR, I'm yet to confirm I can login to Azure and use my authorization code flow token to access Azure OpenAI, I did a few tries yesterday, but to no avail.
> 
> @csotiriou If you know what scope I should set [here](https://github.com/quarkiverse/quarkus-langchain4j/pull/634/files#diff-2f75469ca5e3e4495cd9e80c22d1b620177f7af6fad97896466c301955f8ecadR6) to make it work then let me know please :-)

Unfortunately, I'm not sure, since I'm using IAM and SSO to login and retrieve a token, and the token is vastly different.

Make sure you are providing the Authorization header including the `Bearer ` string, not just the token. 

Is there any error you are getting by using this PR, or something similar? Also, may I ask about your use case? Can I set up a similar environment using a personal account or something similar for testing purposes?

**Comment by sberyozkin** [comment]
Hi @csotiriou Sure, we can finalize the setup for OIDC flows once your PR goes in. (Note the same though would have to work for any provider like vertex-gemini and other similar providers passing tokens, I have a PR opened a bit earlier which works - which is why, should we depend for OIDC on your new interface, it should really be a global one...)

I'm getting infamous `"Unauthorized. Access token is missing, invalid, audience is incorrect (https://cognitiveservices.azure.com), or have expired."`.
 
quarkus-langchain4j setup is in this PR, if you look at the `samples/secure-azure-openai-poem`, in `application.properties`.
I've tried setting a `scope` extra param there to `https://cognitiveservices.azure.com`, `https://cognitiveservices.azure.com/.default`. Apparently someone made it work for the client credentials grant, but I haven't seen anyone confirming it for the authorization code flow, unless there are some indirect confirmations.

I submitted a form to request adding Azure AI service to my Azure tenant, and when it was approved I created an Azure OpenAI account, created a new deployment and there I created a `quarkus-langchain4j` resource. There I added myself as  an `AI Service Developer` to this resource, even though I'm already an owner of it.

The demo flow is that I login to the Quarkus endpoint with Azure and call a poem resource from the HTML page and the token now flows to Azure and the above error is returned. 

If it can be of interest  then indeed, it would be great if you could try this PR with your tenant if you have Azure AI service enabled there. Please take your time in any case, I'm off to my PTO from early morning tomorrow and be back on 11th June, so I won't be able to reply until then.

In any case, thanks for your thoughts, it is appreciated, thanks

**Comment by sberyozkin** [comment]
Also CC Eric, @edeandrea

**Comment by geoand** [comment]
I think this makes sense, but I also think that it should be built on top of https://github.com/quarkiverse/quarkus-langchain4j/pull/646

File: samples/secure-azure-openai-poem/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Secure Vertex AI Gemini Poem Demo
```
Comment (geoand) [inline samples/secure-azure-openai-poem/README.md:1]: Wrong title :)

**Comment by sberyozkin** [comment]
Closing it, will re-open another draft PR

**Pull request closed without merging**